monkestation/code/modules/holomaps/machinery.dm

```diff
@@ -262,6 +262,20 @@
 	if(length(z_transitions))
 		legend += z_transitions
 
+	if(length(GLOB.meteor_shielded_turfs))
+		var/icon/canvas = icon(HOLOMAP_ICON, "blank")
+		var/z_has_coverage = FALSE
+		for(var/turf/open/shielded_turf as anything in GLOB.meteor_shielded_turfs)
+			if(shielded_turf?.z != current_z_level)
```

Comment:
it might be worth turning meteor_shielded_turfs into an assoc list of turfs sorted by their z as that could save on a lot of checks here

Reply:
to be fair, this code is *not* a hot path